src/pages/workspaces/workspace/Data.js

```diff
@@ -563,17 +564,35 @@ const WorkspaceData = _.flow(
     }
   }
 
-  const loadWdsSchema = async () => {
-    if (isFeaturePreviewEnabled('workspace-data-service') && !getConfig().isProd) {
-      try {
-        setWdsSchema([])
-        setWdsSchemaError(undefined)
-        const wdsSchema = await Ajax(signal).WorkspaceData.getSchema(workspaceId)
-        setWdsSchema(wdsSchema)
-      } catch (error) {
-        setWdsSchemaError(error)
+  const loadWdsSchemaAndUrl = async () => {
+    await Ajax(signal).Apps.getV2AppInfo(workspaceId).then(async apps => {
+      // TODO: Update to pull specifically from /api/apps/v2/{workspaceId}/{appName}
+      if (isFeaturePreviewEnabled('workspace-data-service') && !getConfig().isProd) {
+        try {
+          setWdsSchema([])
+          setWdsSchemaError(undefined)
+          const url = await getWdsUrl(apps)
+          setWdsProxyUrl(url)
+          const wdsSchema = await Ajax(signal).WorkspaceData.getSchema(url, workspaceId)
+          setWdsSchema(wdsSchema)
+        } catch (error) {
+          setWdsSchemaError(error)
+        }
       }
+    })
+  }
+
+  // TODO: Do we have the appName here yet? If so, we could just call /api/apps/v2/{workspaceId}/{appName}?
```

Comment:
We'll need to revisit this once we know how'd we like to handle the `appName`

Otherwise the logic in this function might be incomplete?

Reply:
I think you could make a decision now about how to handle the appName, and then just whoever does the create-instance ticket would want to comply.  Sorting by creating date is a backup in case somehow there's more than one by the same name.

Reply:
agreed with making a decision on app name and letting follow-on work adhere to your decision. For now, remember that CBAS and WDS are bundled, so the app name should be something generic that covers both, not just "wds". 

As far as choosing one app in cases where there are multiple … the main criteria here is stability. The UI shouldn't suddenly change which app it is looking at just because the user deployed a second app. (though, I realize this is kinda impossible if the first app is stopped but the second is running)

Reply:
Going to make a follow-up JIRA ticket here!